[AY1920S1-CS2113T-T12-3] BakingHome

liujiajun opened the pull request:
@chekjun
@eugenelim97
@liushiru
@liujiajun

RubaP commented on src/main/java/duke/commons/core/LogsCenter.java:
```diff
@@ -0,0 +1,102 @@
+        Logger logger = Logger.getLogger(name);
+        logger.setUseParentHandlers(false);
+
+        removeHandlers(logger);
```
Method comment says it creates a logger. But why does getLogger deal with Handler?

RubaP commented on src/main/java/duke/commons/core/Message.java:
```diff
@@ -0,0 +1,16 @@
+    public static final String MESSAGE_INVALID_NUMBER_FORMAT = "Number format is invalid.";
+    public static final String MESSAGE_INVALID_INDEX = "Index is invalid.";
+    public static final String MESSAGE_INVALID_RANGE = "Range is invalid.";
+    public static final String MESSAGE_INVALID_STATUS = "Status is invalid.";
```
Good that the messages are declared as final

RubaP commented on src/main/java/duke/commons/util/StringUtil.java:
```diff
@@ -0,0 +1,78 @@
+     * @param sentence cannot be null
+     * @param word     cannot be null, cannot be empty, must be a single word
+     */
+    public static boolean containsWordIgnoreCase(String sentence, String word) {
```
name of a boolean method should be boolean indicative

RubaP commented on src/main/java/duke/commons/util/StringUtil.java:
```diff
@@ -0,0 +1,78 @@
+     * Returns true if {@code s} represents a non-zero unsigned integer
+     * e.g. 1, 2, 3, ..., {@code Integer.MAX_VALUE} <br>
+     * Will return false for any other non-null string input
+     * e.g. empty string, "-1", "0", "+1", and " 2 " (untrimmed), "3 0" (contains whitespace), "1 a" (contains letters)
```
Good comment with very detailed explanations

RubaP commented on src/main/java/duke/logic/CommandManager.java:
```diff
@@ -0,0 +1,78 @@
+//package duke.logic;
```
Why has the entire class commented out? Please delete the file if it not necessary

RubaP commented on src/main/java/duke/logic/LogicManager.java:
```diff
@@ -0,0 +1,107 @@
+        this.storage = storage;
+        this.bakingHomeParser = new BakingHomeParser();
+        this.autoCompleter = new AutoCompleter();
+        addFieldsToAutoComplete();
```
Again the violation of SLAP and SRP. If the responsibility of the constructor is to create the manager with necessary files why it is dealing with adding fields for auto-completion?

RubaP commented on src/main/java/duke/logic/command/CommandResult.java:
```diff
@@ -0,0 +1,70 @@
+        SALE,
+        SAME;
+
+        //public DisplayedPage.PRODUCT getProductPage() {
```
Please remove commented code

RubaP commented on src/main/java/duke/logic/command/CommandResult.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    public enum DisplayedPage {
+        PRODUCT,
```
Good that the constant information is clubbed and declared as enum

RubaP commented on src/main/java/duke/logic/command/decrypted/EditSale.java:
```diff
@@ -0,0 +1,114 @@
+//package duke.logic.command.decrypted;
```
Why does all the Sale related command class are commented out

RubaP commented on src/main/java/duke/logic/command/shopping/BuyShoppingCommand.java:
```diff
@@ -0,0 +1,66 @@
+                Double addedQuantity = toBuy.getQuantity().getNumber();
+                Integer inventoryIndex = inventoryList.indexOf(toBuy);
+                Double currentQuantity = inventoryList.get(inventoryIndex).getQuantity().getNumber();
+                Double totalQuantity = currentQuantity + addedQuantity;
```
How about having a reusable method to compute totalQuantity?

RubaP commented on src/main/java/duke/logic/parser/commons/BakingHomeParser.java:
```diff
@@ -0,0 +1,81 @@
+            subCommandAndArgs = "";
+        }
+
+        switch (primaryCommand) {
```
Good job

RubaP reviewed (commented):
Overall it is a good job with respect to code quality. Kudos to the entire team

liushiru committed:
add CommandTest

liushiru committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-T12-3/main

Commit:
Changed error message as tester asked.

liujiajun committed:
Update README.md

Commit:
Fixed null exception error which occur due to no handling. (#257)

* Removed old command parser and parser.
Tidied up SalePage and SaleCard GUI for inclusion of date filter.

* Added classes to create Sale from completing Orders and buying Ingredients.

* Added method to Complete Order to pass to Sale under Model and Model Manager.

* Reverted spelling change to FIndOrderCommand because merge conflict.

* Merge https://github.com/AY1920S1-CS2113T-T12-3/main

* Merge from https://github.com/AY1920S1-CS2113T-T12-3/main.git

* Merge upstream

* Remove Sale index bug fixed.

* Removed filter on salepage

* Added an AddSaleFromShopping method which creates sale from completing a purchase.

* Sales now accepts negative numbers. Add a -spend true to mark sales as expenditure.
Revenue Cost Profit now updates accordingly.

* Fixed corrupted data because of merge.

* Fixed code quality.

* Optimized imports.

* Added Sale Edit command.

* Added date filter to sale.

* Fixed code quality.

* Fixed missing import.

* Add show sale command.

* Code quality

* Fixed conflict.

* Fixed a bug where double negative makes cost become profit.

* Added Index to SaleCard

* Can delete multiple sale at once

* Used the data file on github.

* sale show now resets filter to show all sale and shows the correct number.

* "sale --" no longer brings user to sale page.

* Exception checking for wrong value and booleans now in place. Date is mm/dd/yyyy as of now.

* Fixed null exception error for filter.

* Changed error message as tester asked.

liujiajun committed:
Temporarily remove shortcut for PE

liujiajun committed:
Update TimeParser

liujiajun committed:
Use formal dates

Commit:
Merge pull request #258 from liushiru/master

AddProductCommandTest

Commit:
Merge branch 'master' into master

Commit:
Merge pull request #259 from liujiajun/master

Remove pretty time, use formal dates

Commit:
Merge https://github.com/AY1920S1-CS2113T-T12-3/main

liushiru committed:
add ProductCommandUtilTest.java

liushiru committed:
sync

liushiru committed:
Merge pull request #261 from liushiru/master

Add ProductCommandUtilTest

liushiru committed:
Add SortProductCommand

liushiru committed:
add Logger

liushiru committed:
add Singleton Pattern

liushiru committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-T12-3/main

liushiru committed:
ignore data file

liushiru committed:
ignore data file

liushiru committed:
build

liushiru committed:
code quality

liujiajun committed:
Use formal dates

liujiajun committed:
Use formal dates (#264)

liushiru committed:
resolve conflict in the data file

liushiru committed:
Merge pull request #262 from liushiru/master

Add logger statements, Singleton, SortProductCommand

liujiajun committed:
Disallow input for re

liujiajun committed:
Disallow input for re

liujiajun committed:
Add shortcut tests

liujiajun committed:
Update docs for TimeParser

liujiajun committed:
Update DG

Commit:
exception handling for shop edit

Commit:
changed error messages

Commit:
added shopping model stub

liujiajun committed:
Fix mass operation bug

liushiru committed:
add getActiveProductsFunction

liushiru committed:
build

liushiru committed:
test

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	data/baking.json

liushiru committed:
change ListProductCommand to FilterProductCommand

liujiajun committed:
Disallow input for-re (#266)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

liushiru committed:
Sync

liushiru committed:
Merge pull request #268 from liushiru/master

add getActiveProductsFunction

Commit:
added some tests, removed stub

Commit:
codacy

eugenelim97 committed:
Merge pull request #267 from eugenelim97/master

Debugging

liushiru committed:
change OrderModelStub to ModelStub

liushiru committed:
Merge pull request #270 from liushiru/master

change OrderModelStub to ModelStub

liujiajun committed:
Add tests; more OOP for order

liujiajun committed:
Add tests

liujiajun committed:
Integrate demo file

liujiajun committed:
Update AutoCompleter

liushiru committed:
IngredientItemList test

liushiru committed:
IngredientItemList test

liujiajun committed:
Update Order

liujiajun committed:
Remove unused commands

liushiru committed:
IngredientItemListParser Test

liushiru committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-T12-3/main

liushiru committed:
sync

liushiru committed:
sync

Commit:
AddShoppingCommandParser Test

liujiajun committed:
Update AboutUs.adoc

liushiru committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-T12-3/main

liushiru committed:
add AddProductIntegrationTest

liujiajun committed:
Clearer UI code

liushiru committed:
complete IngredientParserTest

liushiru committed:
comment out checkstyle

liushiru committed:
code quality

liushiru committed:
Merge pull request #272 from liushiru/master

add test

liushiru committed:
add LogicManagerTest

liushiru committed:
fix some check style

liushiru committed:
Merge pull request #273 from liushiru/master

fix some check style

Commit:
Merge https://github.com/AY1920S1-CS2113T-T12-3/main

Commit:
Added model class diagram.

Commit:
Added DG images

liujiajun committed:
Code quality

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	src/main/java/duke/logic/command/CommandResult.java
#	src/test/java/duke/logic/command/product/AddProductCommandTest.java

Commit:
Update dev guide

liujiajun committed:
Update diagram

liujiajun committed:
Add demo in jar (#271)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

* Add tests; more OOP for order

* Integrate demo file

* Update AutoCompleter

* Update Order

* Remove unused commands

* Clearer UI code

* Code quality

* Update diagram

Commit:
added use case

Commit:
sync

liujiajun committed:
Update DG

eugenelim97 committed:
Updated Dev guide  (#274)

* AddShoppingCommandParser Test

* added use case

liujiajun committed:
Merge branch 'master' into master

liujiajun committed:
DG (#275)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

* Add tests; more OOP for order

* Integrate demo file

* Update AutoCompleter

* Update Order

* Remove unused commands

* Clearer UI code

* Code quality

* Update diagram

* Update DG

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	docs/DeveloperGuide.adoc

liujiajun committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	docs/DeveloperGuide.adoc

liujiajun committed:
DG (#276)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

* Add tests; more OOP for order

* Integrate demo file

* Update AutoCompleter

* Update Order

* Remove unused commands

* Clearer UI code

* Code quality

* Update diagram

* Update DG

Commit:
added manual testing

liujiajun committed:
Change error message

eugenelim97 committed:
Merge branch 'master' into master

liujiajun committed:
Change error message (#280)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

* Add tests; more OOP for order

* Integrate demo file

* Update AutoCompleter

* Update Order

* Remove unused commands

* Clearer UI code

* Code quality

* Update diagram

* Update DG

* Change error message

Commit:
Changed model diagram image

Commit:
Changed Model Diagram Image

Commit:
Update DeveloperGuide.adoc

Commit:
Update DeveloperGuide.adoc

Commit:
Merge changes

eugenelim97 committed:
Merge pull request #279 from eugenelim97/master

Added manual testing

Commit:
merge conflicts

Commit:
Update DG

Commit:
Update model diagram

Commit:
Update DeveloperGuide.adoc

Commit:
Merge branch 'master' into master

Commit:
Merge pull request #281 from chekjun/master

Update DG

liushiru committed:
update UiComponent DG

Commit:
remarks constraints

Commit:
name constraints

liushiru committed:
add FilterProduct implementation

liushiru committed:
update use case

liushiru committed:
add user story and a nfr

liushiru committed:
sync

Commit:
Delete ModelClassDiagram.png

Commit:
Add files via upload

liushiru committed:
update DG

liushiru committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-T12-3/main

Commit:
Delete ModelClassDiagram.png

Commit:
Add files via upload

Commit:
Merge pull request #282 from liushiru/master

update DG

Commit:
Reupload ui.png

liushiru committed:
build

liushiru committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-T12-3/main

Commit:
Delete ModelClassDiagram.png

Commit:
Add files via upload

Commit:
Fixed wrong import causing build fail.

Commit:
Merge https://github.com/AY1920S1-CS2113T-T12-3/main

Commit:
Fixed wrong import causing build fail. (#285)

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	data/baking.json
#	docs/DeveloperGuide.adoc

liujiajun committed:
Update UML (#286)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

* Add tests; more OOP for order

* Integrate demo file

* Update AutoCompleter

* Update Order

* Remove unused commands

* Clearer UI code

* Code quality

* Update diagram

* Update DG

* Change error message

liujiajun committed:
Fix architecture diagram

liujiajun committed:
... (#287)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

* Add tests; more OOP for order

* Integrate demo file

* Update AutoCompleter

* Update Order

* Remove unused commands

* Clearer UI code

* Code quality

* Update diagram

* Update DG

* Change error message

* Fix architecture diagram

liujiajun committed:
不想写了好困

liushiru committed:
update Ui Diagram

liushiru committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-T12-3/main

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	docs/images/storage_diagram.png

liujiajun committed:
好困 (#288)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

* Add tests; more OOP for order

* Integrate demo file

* Update AutoCompleter

* Update Order

* Remove unused commands

* Clearer UI code

* Code quality

* Update diagram

* Update DG

* Change error message

* Fix architecture diagram

* 不想写了好困

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	docs/images/storage_diagram.png

liujiajun committed:
Update UML (#290)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

* Add tests; more OOP for order

* Integrate demo file

* Update AutoCompleter

* Update Order

* Remove unused commands

* Clearer UI code

* Code quality

* Update diagram

* Update DG

* Change error message

* Fix architecture diagram

* 不想写了好困

* Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	docs/images/storage_diagram.png

liushiru committed:
Update Ui diagram (#289)

* build

* update Ui Diagram

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

liujiajun committed:
Change text color

liujiajun committed:
Change text color (#291)

* Use formal dates

* Disallow input for re

* Disallow input for re

* Add shortcut tests

* Update docs for TimeParser

* Update DG

* Fix mass operation bug

* Add tests; more OOP for order

* Integrate demo file

* Update AutoCompleter

* Update Order

* Remove unused commands

* Clearer UI code

* Code quality

* Update diagram

* Update DG

* Change error message

* Fix architecture diagram

* 不想写了好困

* Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	docs/images/storage_diagram.png

* Change text color

liujiajun committed:
Fix indentation

Commit:
edited ingredientbuilder test

Commit:
sync

Commit:
format dev guide + grammar + missing links

liushiru committed:
update ui_class_diagram.png

liushiru committed:
update ui_class_diagram.png

liushiru committed:
change delete product error message

liushiru committed:
changed ShowProductCommand to preamble

liushiru committed:
change edit product command to preamble

liushiru committed:
Merge pull request #293 from liushiru/master

change product error message and -i to preamble

liushiru committed:
add product name constrain

liushiru committed:
add product name constrain (#294)

liujiajun committed:
A few bug fixes

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	data/baking.json

liujiajun committed:
Fix test fail

Commit:
sync

Commit:
travis

eugenelim97 committed:
Merge pull request #292 from eugenelim97/master

Added constraints for name and remarks

eugenelim97 committed:
Merge branch 'master' into master

eugenelim97 committed:
Merge pull request #295 from liujiajun/master

Bug fixes

liujiajun committed:
Update DG

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	docs/DeveloperGuide.adoc

liujiajun committed:
Update DG (#296)

Commit:
Merge https://github.com/AY1920S1-CS2113T-T12-3/main

Commit:
Updated DG

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

Commit:
Merge pull request #297 from chekjun/master

Updated DG

liushiru committed:
update DG

liujiajun committed:
Read storage bug fix

liujiajun committed:
Fix storage bug (#298)

* Update DG

* Read storage bug fix

liushiru committed:
update dg

liujiajun committed:
Update DG

liujiajun committed:
Merge branch 'master' into master

liujiajun committed:
DG (#299)

* Update DG

* Read storage bug fix

* Update DG

liushiru committed:
update FilterProductSD

Commit:
added seq diagram for shop buy

Commit:
'sync'

eugenelim97 committed:
Merge pull request #301 from eugenelim97/master

Sequence diagram for Shop buy

liujiajun committed:
Update DeveloperGuide.adoc

Commit:
added upper limit of ingredient cost of $10000

liushiru committed:
compile CRUD operation for testing manual

liushiru committed:
synce

liujiajun committed:
Fix sc bug

liushiru committed:
Compile CRUD operation for Testing Manual (#302)

* add product name constrain

* update DG

* update dg

* update FilterProductSD

* compile CRUD operation for testing manual

liushiru committed:
resolve conflict

liushiru committed:
resolve conflict

liushiru committed:
synce

liushiru committed:
synce

liushiru committed:
resolve conflict

liujiajun committed:
Merge remote-tracking branch 'origin/master'

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	data/baking.json
#	docs/DeveloperGuide.adoc

liujiajun committed:
Fix Shortcut bugs  (#303)

* Update DG

* Read storage bug fix

* Update DG

* Fix sc bug

liujiajun committed:
Some checkstyle fix

liushiru committed:
update java doc

liushiru committed:
resolve conflict

liushiru committed:
resolve conflict

liushiru committed:
checkstyle

liushiru committed:
Merge pull request #304 from liushiru/master

Fix format for testing manual

Commit:
eugene ppp

Commit:
show on github

Commit:
ppp

Commit:
test

liujiajun committed:
Code quality

liujiajun committed:
Merge branch 'master' of https://github.com/CS2113T-T12-3/main

# Conflicts:
#	data/baking.json
#	docs/DeveloperGuide.adoc
#	src/main/java/duke/commons/util/StringUtil.java

liujiajun committed:
Fix index inconsistant message

Commit:
Merge pull request #305 from liujiajun/master

Code quality; Fix index bug

liujiajun committed:
Update DG

liujiajun committed:
Update DG (#306)

liushiru committed:
sync

liujiajun committed:
Add archived product to demo

Commit:
DG

eugenelim97 committed:
Merge pull request #307 from eugenelim97/master

DG, change upper limit of ingredient cost to 10000

Commit:
DG

liushiru committed:
Merge pull request #308 from eugenelim97/master

DG

liushiru committed:
add ppp

liushiru committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-T12-3/main

liujiajun committed:
Update DG

Commit:
Merge pull request #309 from liujiajun/master

DG

Commit:
added some tests

liujiajun committed:
Update DG

Commit:
DG

Commit:
DG

Commit:
added docs for commands

Commit:
Merge pull request #310 from eugenelim97/master

added some tests

Commit:
Merge pull request #311 from liujiajun/master

Update DG

Commit:
Merge pull request #312 from chekjun/master

Update DG to include sale implementation

Commit:
added documentation

Commit:
Update DeveloperGuide.adoc

Added a missing `s`.

Commit:
Merge pull request #313 from eugenelim97/master

Added documentation for shopping and inventory

Commit:
Update DeveloperGuide.adoc

liushiru committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113T-T12-3/main

liushiru committed:
update DG

liushiru committed:
update DG

liushiru committed:
update DG

liujiajun committed:
Add JavaDoc (#314)

* Update DG

* Update DG

* JavaDoc

liujiajun committed:
Update Ui.png (#315)

* Update DG

* Update DG

* JavaDoc

* Update Ui

liujiajun committed:
Update UserGuide.md

liujiajun committed:
Update README.md

liujiajun committed:
Update UserGuide.md

liujiajun committed:
Update README.md

liujiajun committed:
Update README.md

liujiajun committed:
Update README.md

liujiajun committed:
Update README.md

liujiajun committed:
Update README.md

eugenelim97 committed:
Update README.md

liujiajun committed:
Update README.md
